iceoryx_posh/test/moduletests/test_popo_client_port.cpp

```diff
@@ -0,0 +1,182 @@
+        clientPortRouDi.dispatchCaProMessageAndGetPossibleResponse(serverMessage);
+        ASSERT_THAT(clientPortData.m_connectionState, Eq(iox::ConnectionState::CONNECT_HANDSHAKE));
+        if (clientPortData.m_connectionState == state)
+        {
+            return;
+        }
+
+        serverMessage = iox::capro::CaproMessage{
+            iox::capro::CaproMessageType::ACK, m_serviceDescription, iox::capro::CaproMessageSubType::NOSUBTYPE};
+        clientPortRouDi.dispatchCaProMessageAndGetPossibleResponse(serverMessage);
+        ASSERT_THAT(clientPortData.m_connectionState, Eq(iox::ConnectionState::CONNECTED));
+        if (clientPortData.m_connectionState == state)
+        {
+            return;
+        }
+
+        constexpr bool NOT_IMPLEMENTED{true};
+        ASSERT_FALSE(NOT_IMPLEMENTED);
+    }
+
+    uint32_t getNumberOfUsedChunks() const
+    {
+        return m_memoryManager.getMemPoolInfo(0).m_usedChunks;
+    }
+
+  private:
+    static constexpr iox::units::Duration DEADLOCK_TIMEOUT{5_s};
+    Watchdog m_deadlockWatchdog{DEADLOCK_TIMEOUT};
+
+    static constexpr size_t MEMORY_SIZE = 1024 * 1024;
+    uint8_t m_memory[MEMORY_SIZE];
+    iox::posix::Allocator m_memoryAllocator{m_memory, MEMORY_SIZE};
+    iox::mepoo::MemoryManager m_memoryManager;
+
+    iox::capro::ServiceDescription m_serviceDescription{"hyp", "no", "toad"};
+    iox::RuntimeName_t m_runtimeName{"hypnotoad"};
+
+    ClientOptions m_withConnectOnCreate = [] {
+        ClientOptions options;
+        options.connectOnCreate = true;
+        return options;
+    }();
+    ClientOptions m_withoutConnectOnCreate = [] {
+        ClientOptions options;
+        options.connectOnCreate = false;
+        return options;
+    }();
+
+  public:
+    static constexpr uint32_t USER_PAYLOAD_SIZE{32};
+    static constexpr uint32_t USER_PAYLOAD_ALIGNMENT{8};
+
+    ServerChunkQueueData_t serverChunkQueueData{iox::popo::QueueFullPolicy::DISCARD_OLDEST_DATA,
+                                                iox::cxx::VariantQueueTypes::SoFi_MultiProducerSingleConsumer};
+
+    // client port with connect on create
+    ClientPortData clientPortDataWithConnectOnCreate{
+        m_serviceDescription, m_runtimeName, m_withConnectOnCreate, &m_memoryManager};
+    ClientPortUser clientPortUserWithConnectOnCreate{clientPortDataWithConnectOnCreate};
+    ClientPortRouDi clientPortRouDiWithConnectOnCreate{clientPortDataWithConnectOnCreate};
+
+    // client port without connect on create
+    ClientPortData clientPortDataWithoutConnectOnCreate{
+        m_serviceDescription, m_runtimeName, m_withoutConnectOnCreate, &m_memoryManager};
+    ClientPortUser clientPortUserWithoutConnectOnCreate{clientPortDataWithoutConnectOnCreate};
+    ClientPortRouDi clientPortRouDiWithoutConnectOnCreate{clientPortDataWithoutConnectOnCreate};
+};
+constexpr iox::units::Duration ClientPort_test::DEADLOCK_TIMEOUT;
+
+TEST_F(ClientPort_test, InitialConnectionStateOnPortWithConnectOnCreateIs_CONNECTED)
```

Comment:
Just in case you are wondering, the tests are currently WIP and will be extended. Please no comments regarding missing tests for now.